### ISSUE TITLE
indent-error-flow reports a false positive

### ISSUE THREAD

**Issue by mgechev** [mgechev/revive#19]
**Describe the bug**

`indent-error-flow` gives false positive for:

```go
func h(f func() bool, x int) string {
	if err == author.ErrCourseNotFound {
		return
	} else if err == author.ErrCourseAccess {
		// side effect
	} else if err == author.AnotherError {
		return "okay"
	} else {
		// side effect
	}
}
```

**To Reproduce**
1. I updated revive `go get -u github.com/mgechev/revive`
2. I run the rule `indent-error-flow`
3. I got a warning `if block ends with a return statement, so drop this else and outdent its block`

**Expected behavior**

Should not get a warning in the case above because of the `else if` statements.

**Comment by mgechev** [comment]
@chavacava let me know if you want to take this one.

**Comment by chavacava** [comment]
Hi, please take a look to the corresponding pull request.
